### PR TITLE
Move files of utilities in the testkit package to the testkit project

### PR DESCRIPTION
**Pull request by lrytz**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -645,7 +645,7 @@ lazy val bench = project.in(file("test") / "benchmarks")
 
 
 lazy val testkit = configureAsSubproject(project)
-  .dependsOn(library)
+  .dependsOn(compiler)
```
Comment (SethTisue) [inline build.sbt:648]: why?

**Comment by SethTisue** [comment]
previous history is in #7817 + #7847 

there is a test failure, reproducible locally:

```
sbt:root> junit/testOnly *SourceTest
...
[error] Failed tests:
[error] 	scala.io.SourceTest
```

I think it's caused by this change

```diff
-  .dependsOn(library)
+  .dependsOn(compiler)
```

because the test picks up the compiler `rootdoc.txt` instead of the stdlib one

File: build.sbt (hunk 1 above)
Comment (lrytz) [reply build.sbt:648]: The moved files depend on the compiler, for example https://github.com/scala/scala/blob/b81be74f072a19c446e1c17a24dfacc4d69f3062/src/testkit/scala/tools/testkit/BytecodeTesting.scala creates a Global

Comment (SethTisue) [reply build.sbt:648]: ah, okay. I see that everybody who depends on `testkit` is already also depending on `compiler`, so no worries.

**Review by SethTisue — APPROVED** [review]
LGTM for merge once test failure is addressed

**Pull request merged**